packages/cuba-rest-js/src/event_emitter.ts

```diff
@@ -0,0 +1,58 @@
+interface IKeyFuncVal {
+    [k: string]: (...args) => any;
+}
+
+class EventEmitter<Events extends IKeyFuncVal, Key extends keyof Events = keyof Events> {
```

Comment:
I'm not sure we need that much typings complexity. Even for generic case. For example, I can't think of a realistic situation when we need `Events` to be something else than `IKeyFuncVal`. I'd simplify it to make code more readable / easier to follow. Also by doing so we might get rid of `@ts-ignore` below.

Perhaps we could leave it as it is for now but create a separate ticket.

Reply:
@web-devel WDYT?

Reply:
agree